#159987709 Confugure PostgreSQL As DB Engine

faksam opened the pull request:
#### What does this PR do?
- Confugure PostgreSQL & Sequelize ORM
- create .env.sample
- create .sequelizerc
- create sequelize configurations
- create migrations
- modify user models file to use sequelize

#### Description of Task to be completed?
- modify passport configuration to use postgres
- 

#### How should this be manually tested?
- Clone the repo
- Run `npm install`
- Run `npm run migration`

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- [#159987709](https://www.pivotaltracker.com/story/show/159987709)

eshaibu commented on models/User.js:
```diff
@@ -1,57 +1,37 @@
+    },
+  });
+  return User;
+};
```
- Create a `beforeCreate` hook for hashing passwords before saving
- Create an Instance `toAuthJSON` to  mimic what was there in the mongoose model

eshaibu commented on routes/api/users.js:
```diff
@@ -1,21 +1,49 @@
+        }
+      })
+      .then(user => res.status(201).send(user))
+      .catch(error => res.status(400).send(error));
```
Maintain the  `.catch(next)`

eshaibu reviewed (changes requested):
- You removed the package-lock.json file, add it back its important
- Take out mongoose connection in index.js at the root directory
- Add installation steps to the readme file
Check the other comments made and impement changes

faksam committed:
chore(models): configure postgresql

- setup postgresql
- setup sequelize
- create .env.sample
- create .sequelizerc
- create sequelize configurations
- create migrations
- modify user models file to use sequelize

[Finishes #159987709]

eshaibu commented on migrations/20180828205453-user.js:
```diff
@@ -0,0 +1,39 @@
+        type: Sequelize.UUID,
+        defaultValue: Sequelize.UUIDV4
+      },
+      hash: {
```
Simply `password`

eshaibu commented on models/User.js:
```diff
@@ -1,57 +1,55 @@
+      type: DataTypes.STRING,
+    },
+  },
+  {
```
We are using sequelize 4 so use this something like
```
User.hook('beforeCreate', (newUser) => {
    newUser.password = bcrypt.hashSync(newUser.password, bcrypt.genSaltSync(8));
});
```

eshaibu commented on models/User.js:
```diff
@@ -1,57 +1,55 @@
+        newUser.confirmPassword = bcrypt.hashSync(newUser.confirmPassword, bcrypt.genSaltSync(8));
+      }
+    },
+    instanceMethods: {
```
Use 
```
User.prototype.toAuthJSON = function () {
  return .....
}
```
